lapce-core/src/selection.rs

```diff
@@ -50,22 +53,48 @@ impl SelRegion {
         self.min() <= offset && offset <= self.max()
     }
 
+    /// Return the minimum value between region's start and end position
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// # fn main() {
+    /// # use lapce_core::selection::SelRegion;
+    /// let  region = SelRegion::new(1, 10, None);
+    /// assert_eq!(region.min(), region.start);
+    /// let  region = SelRegion::new(42, 1, None);
+    /// assert_eq!(region.min(), region.end);
+    /// # }
     pub fn min(self) -> usize {
         min(self.start, self.end)
     }
 
+    /// Return the maximum value between region's start and end position.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// # fn main() {
+    /// # use lapce_core::selection::SelRegion;
+    /// let  region = SelRegion::new(1, 10, None);
+    /// assert_eq!(region.max(), region.end);
+    /// let  region = SelRegion::new(42, 1, None);
+    /// assert_eq!(region.max(), region.start);
+    /// # }
     pub fn max(self) -> usize {
         max(self.start, self.end)
     }
 
-    pub fn start(self) -> usize {
-        self.start
-    }
-
-    pub fn end(self) -> usize {
-        self.end
-    }
-
+    /// A [`SelRegion`] is considered to be a caret when its start and end position are equal.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// # fn main() {
```

Comment:
I think you don't need to wrap this in `fn main` if there's no return value (and no usage of `?`).